src/utils/transforms.cljs

```diff
@@ -33,15 +33,16 @@
   ([^js m]
    (<-js-map m nil))
   ([^js m {:keys [key-fn val-fn]}]
-   (when (identical? (type m) js/Object)
+   (if (identical? (type m) js/Object)
      (persistent!
       (reduce (fn [r k]
                 (let [v       (oops/oget+ m k)
                       new-key (if key-fn (key-fn k v) k)
                       new-val (if val-fn (val-fn k v) v)]
                   (assoc! r new-key new-val)))
               (transient {})
-              (js-keys m))))))
+              (js-keys m)))
+     m)))
```

Comment:
If the function is called with anything besides a Clojure map or a JS Object I think it should return nil, better would be to fail an instrumentation check with malli. Returning the input itself would probably be masking a more serious bug in the code, because why would a developer pass a different data type than a map or JS object to this function?

We could be more strict about the data type. Since this function is multi-arity and our instrumentation utilities don't work with them, we can do something simple for now. Use `(assert (or (identical? (type m) js/Object) (map? m)))` because shadow-cljs will elide assertions in release builds.

> :elide-asserts (Boolean) default to false in development and true in release builds
>
> -- https://shadow-cljs.github.io/docs/UsersGuide.html#compiler-options

Reply:
Hi @ilmotta, Thank you for suggestion. Added [assert](https://github.com/status-im/status-mobile/pull/21238/commits/f6e5ec0ab7b76089a2ea8c56fd3524f642b315f1), please let me know if it looks ok?

Reply:
Looks good 👍🏼

Reply:
Updated. Thank you